crates/ruma-client-api/src/r0/session/login.rs

```diff
@@ -78,6 +78,7 @@ impl Response {
 /// Identification information for the user.
 #[derive(Clone, Debug, PartialEq, Eq, Outgoing, Serialize)]
 #[serde(from = "user_serde::IncomingUserIdentifier", into = "user_serde::UserIdentifier<'_>")]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
 pub enum UserIdentifier<'a> {
```

Comment:
This type seems extremely unlikely to have additional variants in the future. Probably reasonable to have it exhaustive.